Common++/header/MacAddress.h

```diff
@@ -29,60 +29,69 @@ namespace pcpp
+		 * and the remaining bytes are ignored. If the byte array is invalid, the constructor throws.
 		 * @param[in] addr A pointer to the byte array containing 6 bytes representing the MAC address
 		 */
-		MacAddress(const uint8_t* addr) : m_IsValid(true) { memcpy(m_Address, addr, sizeof(m_Address)); }
+		explicit MacAddress(const uint8_t* addr) { memcpy(m_Address, addr, sizeof(m_Address)); }
 
 		/**
-		 * A constructor that creates an instance of the class out of a (char*) string.
-		 * If the string doesn't represent a valid MAC address, instance will be invalid, meaning isValid() will return false
-		 * @param[in] addr A pointer to the (char*) string
+		 * A constructor that creates an instance of the class out of a std::string.
+		 * If the string doesn't represent a valid MAC address, the constructor throws.
+	 	 * @param[in] addr the string representing the MAC address in format "00:00:00:00:00:00"
 		 */
-		MacAddress(const char* addr) { init(addr); }
+		explicit MacAddress(const std::string& addr);
 
 		/**
-		 * A constructor that creates an instance of the class out of a std::string.
-		 * If the string doesn't represent a valid MAC address, instance will be invalid, meaning isValid() will return false
-	 	 * @param[in] addr A pointer to the string
+		 * A template constructor that creates an instance of the class out of a string convertible to std::string.
+		 * If the string doesn't represent a valid MAC address, the constructor throws.
+		 * @param[in] addr the string representing the MAC address in format "00:00:00:00:00:00"
 		 */
-		MacAddress(const std::string& addr) { init(addr.c_str()); }
+		template<typename T, typename = typename std::enable_if<std::is_convertible<T, std::string>::value>::type>
+		MacAddress(const T& addr) : MacAddress(static_cast<std::string>(addr)) {};
+
 
 		/**
 		 * A constructor that creates an instance of 6 bytes representing the MAC address
-		 * @param[in] firstOctest Represent the first octet in the address
+		 * @param[in] firstOctet Represent the first octet in the address
 		 * @param[in] secondOctet Represent the second octet in the address
 		 * @param[in] thirdOctet Represent the third octet in the address
 		 * @param[in] fourthOctet Represent the fourth octet in the address
 		 * @param[in] fifthOctet Represent the fifth octet in the address
 		 * @param[in] sixthOctet Represent the sixth octet in the address
 		 */
-		inline MacAddress(uint8_t firstOctest, uint8_t secondOctet, uint8_t thirdOctet, uint8_t fourthOctet, uint8_t fifthOctet, uint8_t sixthOctet);
+		inline MacAddress(uint8_t firstOctet, uint8_t secondOctet, uint8_t thirdOctet, uint8_t fourthOctet, uint8_t fifthOctet, uint8_t sixthOctet)
+		{
+			m_Address[0] = firstOctet;
+			m_Address[1] = secondOctet;
+			m_Address[2] = thirdOctet;
+			m_Address[3] = fourthOctet;
+			m_Address[4] = fifthOctet;
+			m_Address[5] = sixthOctet;
+		}
 
 #if __cplusplus > 199711L || _MSC_VER >= 1800
```

Comment:
I think we can remove this `#if` now, as PcapPlusPlus supports C++11

Reply:
fixed